customersupport/controllers/ticket.py

```diff
@@ -1,32 +1,68 @@
 from customersupport import app
-from customersupport.wrappers import sales
-from customersupport.models import Ticket, CallLog
+from customersupport.wrappers import sales, hr
+from customersupport.models import Ticket, CallLog, TicketType, TicketStatus
 from customersupport.database import db_session
-from customersupport.util import get_param_from_request_if_not_empty
+from customersupport.util import get_param_from_request_if_not_empty, get_post_data
 
-from flask import Flask, request, render_template
+from flask import Flask, request, render_template, abort
 from flask import request
 from flask import render_template
 from flask import jsonify
 from flask import json
 
-@app.route('/sales/refund/real')
-def refund_order(mock=False):
-    order = sales.initiate_refund(
-        replace=True,
-        order_id=1,
-        serial_numbers=[
-            123,
-            456,
-            789
-        ],
-        mock=mock
+@app.route('/api/refund', methods=['POST'])
+def refund_order():
+
+    use_mock = get_post_data("use_mock") is not None
+    replace = get_post_data("replace") == "true"
+    ticket_type = TicketType.REFUND
+    if replace:
+        ticket_type = TicketType.REPLACE
+    status = get_post_data("status")
+    current_status = TicketStatus.CLOSED
+    if status == "open":
+        current_status = TicketStatus.OPEN
+    elif status == "pend":
+        current_status = TicketStatus.PENDING
+    elif status == "pass" or status == "fail":
+        current_status = TicketStatus.CLOSED
+    order_id = get_post_data("order_id")
+    serial_ids = get_post_data("serial_ids")
+    customer_id = get_post_data("customer_id")
+
+    ticket = Ticket(
+        ticket_type=ticket_type,
+        customer_id=customer_id,
+        order_id=order_id,
+        current_status=current_status
     )
 
-    if order is not None:
-        return jsonify(order.serialize())
+    db_session.add(ticket)
+    db_session.commit()
+
+    try:
+        status_is_good = sales.initiate_refund(
+            replace=replace,
+            order_id=order_id,
+            serial_numbers=serial_ids,
+            mock=use_mock
+        )
+        if status_is_good:
+            """
+            Yay, we came back successfully, now go tell hr we made a refund/return.
+            Blind fire for now, we do not care if this is delivered or not.
+            TODO: Actually pass an employeeId
+            """
+            hr.report_action(replace, order_id, serial_ids, "1")
+            # Return the ticket created ealier
+            return jsonify(ticket.serialize())
+    except Exception as ex:
+        print(ex)
+        abort(500)
+
     else:
-        return jsonify([])
+        abort(500)
+        return jsonify({})
 
 
 @app.route('/sales/refund/stub')
```

Comment:
Could you also remove the stub version here? It shouldn't be being used at this point.